08-rails/testing-db-in-controllers.md

```diff
@@ -49,21 +63,31 @@ describe "create" do
 
     must_respond_with  :redirect
 
-    expect(Book.last.title).must_equal book_hash[:book][:title]
-    expect(Book.last.author_id).must_equal book_hash[:book][:author_id]
-    expect(Book.last.description).must_equal book_hash[:book][:description]
+    book = Book.find_by(title: "Practical Object Oriented Programming in Ruby")
+
+    expect(book).wont_be_nil
+    expect(book.title).must_equal book_hash[:book][:title]
+    expect(book.author).must_equal book_hash[:book][:author]
+    expect(book.description).must_equal book_hash[:book][:description]
   end
 end
 ```
 
 The example above illustrates that the test can pass in a mock-params hash into the request with the `params: params_hash` argument.  Notice we verify that the number of books in the database increases and that the last book in the database has the correct title, author and description.
 
-**Exercise** You also need to create a test in which the params are invalid, violating validations.  With a partner write another test in which the params are invalid.  You can see an example [here.](code_samples/create_controller_test.rb)
+**Exercise** You also need to create a test in which the params are invalid or missing.  With a partner write another test in which the params are invalid.  We have an [example solution](code_samples/create_controller_test.rb)
 
```

Comment:
If the params are missing, will that cause a problem? Remember that we have no validations for this model.

Reply:
There wont' be a `:book` field in params and strong params should fail causing an exception.  We could still take this test out.

Reply:
Ah, I see. I like the idea then, it'll be good to have a test for a failure case.

Will we have introduced strong params at this point in the week? Looks like we don't cover that until Friday?

Reply:
I'll pull it out for Wednesday and add it back in on Friday.